accepted/2020/platform-checks/platform-checks.md

```diff
@@ -0,0 +1,450 @@
+
+## Design
+
+### Attribute
+
+OS bindings and platforms-specific BCL APIs will be annotated using a set of
+custom attributes:
+
+```C#
+namespace System.Runtime.InteropServices
+{
+    public partial struct OSPlatform
+    {
+        // Existing properties
+        // public static OSPlatform FreeBSD { get; }
+        // public static OSPlatform Linux { get; }
+        // public static OSPlatform OSX { get; }
+        // public static OSPlatform Windows { get; }
+        public static OSPlatform Android { get; }
+        public static OSPlatform iOS { get; }
+    }
+
+    public partial static class RuntimeInformation
+    {
+        // Existing API
+        // public static bool IsOSPlatform(OSPlatform osPlatform);
+
+        // Check for the OS with a >= version comparison
+        // Used to guard APIs that were added in the given OS release.
+        public static bool IsOSPlatformOrLater(OSPlatform osPlatform, int major);
+        public static bool IsOSPlatformOrLater(OSPlatform osPlatform, int major, int minor);
+        public static bool IsOSPlatformOrLater(OSPlatform osPlatform, int major, int minor, int build);
+        public static bool IsOSPlatformOrLater(OSPlatform osPlatform, int major, int minor, int build, int revision);
+
+        // Allows checking for the OS with a < version comparison
+        // Used to guard APIs that were obsoleted or removed in the given OS release.
+        public static bool IsOSPlatformAndEarlier(OSPlatform osPlatform, int major);
+        public static bool IsOSPlatformAndEarlier(OSPlatform osPlatform, int major, int minor);
+        public static bool IsOSPlatformAndEarlier(OSPlatform osPlatform, int major, int minor, int build);
+        public static bool IsOSPlatformAndEarlier(OSPlatform osPlatform, int major, int minor, int build, int revision);
+    }
+}
+
+namespace System.Runtime.Versioning
+{
+    public abstract class OSPlatformAttribute : Attribute
+    {
+        protected OSPlatformAttribute(string osPlatform,
+                                      int major,
+                                      int minor,
+                                      int build,
+                                      int revision);
+        public string PlatformIdentifier { get; }
+        public int Major { get; }
+        public int Minor { get; }
+        public int Build { get; }
+        public int Revision { get; }
+    }
+
+    [AttributeUsage(AttributeTargets.Assembly |
+                    AttributeTargets.Class |
+                    AttributeTargets.Constructor |
+                    AttributeTargets.Event |
+                    AttributeTargets.Method |
+                    AttributeTargets.Module |
+                    AttributeTargets.Property |
+                    AttributeTargets.Struct,
+                    AllowMultiple=false, Inherited=true)]
+    public sealed class AddedInOSPlatformAttribute : OSPlatformAttribute
```

Comment:
I like `Introduced` instead of `Added`. Also having "Version" like previously mentioned seems more accurate, so something like `IntroducedInOSVersionAttribute` sounds better to me (or maybe even something simpler like `IntroducedInAttribute`?)

Reply:
Agreed here, I like "version" in there... else it is like it was added in iOS itself